scripts/generate_vif/generate_vif.py

```diff
@@ -0,0 +1,259 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2022 The Chromium OS Authors
+# SPDX-License-Identifier: Apache-2.0
+
+import argparse
+import os
+import pickle
+import sys
+import xml.etree.ElementTree as ET
+
+import constants
+
+SCRIPTS_DIR = os.path.join(os.path.dirname(__file__), "..")
+sys.path.insert(0, os.path.join(SCRIPTS_DIR, 'dts', 'python-devicetree', 'src'))
+
+from devicetree import edtlib
+
+
+def main():
+    args = parse_args()
+    try:
+        with open(args.edt_pickle, 'rb') as f:
+            edt = pickle.load(f)
+    except edtlib.EDTError as err:
+        sys.exit(f"devicetree error: {err}")
+    xml_root = get_root()
+    add_elements_to_xml(xml_root, constants.VIF_SPEC_ELEMENTS)
+    add_element_to_xml(xml_root, constants.MODEL_PART_NUMBER, args.board)
```

Comment:
Have you considered using the `/model` devicetree property instead? Would save a flag...

Reply:
> Have you considered using the /model devicetree property instead?

Please be careful when using 'model'. Here is a thread with clarifications about its intended usage on the DT spec mailing list you should understand if you want to use this: https://www.spinics.net/lists/devicetree-spec/msg01065.html

Reply:
Interesting, thanks for the pointer Martí, maybe it's better to leave it alone for the moment.

@madhurimaparuchuri is it important to have control over that? May be worth having a Kconfig option for overriding it manually? Then the application could override that option default using something from DT or something else.

Reply:
As per internal discussions, this seems to be not an serious member of XML for now, and the value could be different in future, it will be updated or removed in the next revision. Thanks for the suggestions 😀